src/Elastic.Apm.AspNetCore/ApplicationConfigurationReader.cs

```diff
@@ -4,15 +4,16 @@
 using Elastic.Apm.Logging;
 using Microsoft.Extensions.Configuration;
 
-namespace Elastic.Apm.AspNetCore.Config
+namespace Elastic.Apm.AspNetCore
 {
 	/// <summary>
 	/// An agent-config provider based on Microsoft.Extensions.Configuration.IConfiguration.
-	/// It uses environment variables as fallback
+	/// It uses environment variables as fallback.
 	/// </summary>
-	internal class MicrosoftExtensionsConfig : AbstractConfigurationReader, IConfigurationReader
+	internal class ApplicationConfigurationReader : AbstractConfigurationReader, IConfigurationReader
```

Comment:
`Elastic.Apm.AspNetCore` and `Elastic.Apm.AspNetFullFramework` are corresponding projects, one for .NET Core, one for Full Framework. Therefore, any corresponding classes should follow a similar pattern.

- `MicrosoftExtensionsConfig` extends `AbstractConfigurationReader` and realises `IConfigurationReader`.
- `FullFrameworkConfigReader` extends `EnvironmentConfigurationReader`.

Conventionally, derived classes include all or part of the name of the base class. Most classes in Elastic APM Agent follow this pattern. `FullFrameworkConfigReader` _almost_ follows that pattern but abbreviates `Configuration`. `MicrosoftExtensionsConfig` abbreviates `Configuration` _and_ drops `Reader`. This is incorrect because the class is a reader (a config reader specifically) and is not config. That is also implied by its base class. The class _extends_ the base class.

The fact that it uses `IConfiguration` reader, which simply happens to be in the `Microsoft.Extensions` namespace, should not affect the class name. This is implementation. The class name should reflect what the class _does_, not how it is _implemented_. It is highly unusual to name any class based on the namespace of other classes or interfaces it inherits. It is more usual to name it based on the inherited class or interface name itself.

**Advantage:** The reader can understand what the class does rather than seeing some obscure namespace embedded into the class name.

In any case, I've reverted the change and will put it in a separate PR.